core-api/src/main/java/com/optimizely/ab/OptimizelyUserContext.java

```diff
@@ -0,0 +1,195 @@
+/**
+ *
+ *    Copyright 2020, Optimizely and contributors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+package com.optimizely.ab;
+
+import com.optimizely.ab.optimizelydecision.OptimizelyDecideOption;
+import com.optimizely.ab.optimizelydecision.OptimizelyDecision;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class OptimizelyUserContext {
+    @Nonnull
+    private final String userId;
+
+    @Nonnull
+    private final Map<String, Object> attributes;
+
+    @Nonnull
+    private final Optimizely optimizely;
+
+    private static final Logger logger = LoggerFactory.getLogger(OptimizelyUserContext.class);
+
+    public OptimizelyUserContext(@Nonnull Optimizely optimizely,
+                                 @Nonnull String userId,
+                                 @Nonnull Map<String, Object> attributes) {
+        this.optimizely = optimizely;
+        this.userId = userId;
+        this.attributes = Collections.synchronizedMap(new HashMap<>(attributes));
```

Comment:
This will also throw null pointer exception if attributes value is null. Plus we need to initialize it as new empty hashMap because of set attributes method as it is assuming that attributes is not null.
```suggestion
        if (attributes != null) {
            this.attributes = Collections.synchronizedMap(new HashMap<>(attributes));
        } else {
            this.attributes = Collections.synchronizedMap(new HashMap<>());
        }
```